hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMDirectoriesPurgeRequestWithFSO.java

```diff
@@ -168,7 +175,16 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, TermIn
             volBucketInfoMap.putIfAbsent(volBucketPair, omBucketInfo);
           }
         }
+        deletingServiceMetrics.incrNumSubKeysPurged(numSubFilesDeleted);
+        deletingServiceMetrics.setNumSubKeysPurgedInLatestRequest(numSubFilesDeleted);
+        if (path.hasDeletedDir()) {
+          deletingServiceMetrics.incrNumDirPurged(1);
+          deletingServiceMetrics.setNumDirPurgedInLatestRequest(1);
```

Comment:
If there are multiple paths in request, this will just set 1 or 0..Either you can remove `NumDirPurgedInLatestRequest` metrics or do it after the for loop and set count correctly.